src/main/java/de/tum/cit/aet/artemis/atlas/dto/LearningPathNavigationObjectDTO.java

```diff
@@ -44,10 +45,35 @@ public static LearningPathNavigationObjectDTO of(LearningObject learningObject,
 
         name = unreleased ? "" : name;
 
-        return new LearningPathNavigationObjectDTO(id, completed, name, competencyId, type, unreleased);
+        return new LearningPathNavigationObjectDTO(id, completed, name, competencyId, type, repeatedTest, unreleased);
     }
 
     public enum LearningObjectType {
         LECTURE, EXERCISE
     }
+
+    /**
+     * Check if the learning object represented by this dto is equal to the given learning object in the competency.
+     *
+     * @param learningObject the learning object to compare to
+     * @param competency     the competency to compare to
+     * @return whether the learning object is equal
+     */
+    public boolean equalsLearningObject(LearningObject learningObject, CourseCompetency competency) {
+        if (learningObject == null) {
+            return false;
+        }
+        if (id != learningObject.getId()) {
+            return false;
+        }
+        if (competencyId != competency.getId()) {
+            return false;
+        }
```

Comment:
Would it make sense to put this into one if statement? Or did you choose to spread it out for readability?